presto-main/src/main/java/com/facebook/presto/sql/planner/PlanFragmenter.java

```diff
@@ -282,6 +284,18 @@ private void chooseDistribution(PartitioningHandle distribution)
                     partitioningHandle);
         }
 
+        private boolean isCompatibleSystemPartitioning(PartitioningHandle distribution)
+        {
+            ConnectorPartitioningHandle currentHandle = partitioningHandle.get().getConnectorHandle();
+            ConnectorPartitioningHandle distributionHandle = distribution.getConnectorHandle();
+            if ((currentHandle instanceof SystemPartitioningHandle) &&
+                    (distributionHandle instanceof SystemPartitioningHandle)) {
+                return ((SystemPartitioningHandle) currentHandle).getPartitioning() ==
```

Comment:
I'm not sure certain distributions are compatible in all cases (e.g., fixed-hash vs fixed-arbitrary).